refinery/data_set_manager/utils.py

```diff
@@ -629,7 +632,14 @@ def _generate_solr_params(params, assay_uuids):
         return None
     solr_params = 'fq=assay_uuid:({})'.format(' OR '.join(assay_uuids))
 
-    if facet_field:
+    if facets_from_config:
+        # Twice as many facets as necessary, but easier than the alternative.
```

Comment:
This might take a second for someone to figure out. How about:
```
template = '&facet.field={0}_Characteristics_generic_s' + \
        '&facet.field={0}_Factor_Value_generic_s'
solr_params += ''.join([template.format(s) for s in settings.USER_FILES_FACETS.split(",")])
```